subsys/bluetooth/host/l2cap.c

```diff
@@ -2039,33 +1998,12 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 }
 
 /* return next netbuf fragment if present, also assign metadata */
-static struct net_buf *prepare_next_frag(struct net_buf *parent,
-					 struct l2cap_tx_meta_data *meta)
-{
-	/* this does an unref on `parent` */
-	struct net_buf *next = net_buf_frag_del(NULL, parent);
-
-	if (next) {
-		LOG_DBG("process next frag: %p -> %p", parent, next);
-
-		/* Copy the l2cap metadata ref to the next buffer in the chain.
-		 * We need it to figure out which channel the PDU has been sent
-		 * on, in order to either send the rest of the SDU or call the
-		 * application callback.
-		 */
-		l2cap_tx_meta_data(next) = meta;
```

Comment:
Do you not need to set the `cid` in the buffer user data for the next fragment?

Reply:
nope, because now we don't read from the `buf` userdata from the callback.
We get the CID from the userdata that bt_conn passes to us.
Same confusing name, different storage locations.